lib/puppet/parser/functions/platform_get.rb

```diff
@@ -0,0 +1,54 @@
+#
+# Cisco platform_get puppet manifest function.
+#
+# January, 2016
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Puppet
+  module Parser
+    # Function platform_get.  Returns platform string.
+    module Functions
+      newfunction(:platform_get, type: :rvalue) do |_args|
+        data = lookupvar('cisco')
+        fail Puppet::ParseError,
+             "Facter fact 'cisco' unavailable on this platform\n#{__FILE__}" if data.nil?
+        pi = data['hardware']['type']
```

Comment:
This now generates the following error when the cisco fact is unavailable:

```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Evaluation Error: Error while evaluating a Function Call, Facter fact 'cisco' unavailable on this platform
/etc/puppetlabs/code/environments/production/modules/ciscopuppet/lib/puppet/parser/functions/platform_get.rb at /etc/puppetlabs/code/environments/production/manifests/site.pp:3:18 on node dt-n9k5-1.cisco.com
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run
root@dt-n9k5-1#
```
NOTE: I added `__FILE__` because no path information provided to the offending function.